Akka.Serialization.MessagePack/MsgPackSerializer.cs

```diff
@@ -5,31 +5,26 @@
 //-----------------------------------------------------------------------
 
 using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
 using System.Threading;
 using Akka.Actor;
 using Akka.Configuration;
 using Akka.Serialization.MessagePack.Resolvers;
 using MessagePack;
+using MessagePack.Formatters;
 using MessagePack.ImmutableCollection;
 using MessagePack.Resolvers;
+using Newtonsoft.Json;
 
 namespace Akka.Serialization.MessagePack
 {
     public sealed class MsgPackSerializer : Serializer
     {
-        internal static AsyncLocal<ActorSystem> LocalSystem = new AsyncLocal<ActorSystem>();
         private readonly MsgPackSerializerSettings _settings;
-
-        static MsgPackSerializer()
-        {
-            CompositeResolver.RegisterAndSetAsDefault(
-#if SERIALIZATION
-                SerializableResolver.Instance,
-#endif
-                AkkaResolver.Instance,
-                ImmutableCollectionResolver.Instance,
-                TypelessContractlessStandardResolver.Instance);
-        }
+        private readonly IFormatterResolver _resolver;
```

Comment:
So I'm a tad mystified by the formatter / resolver system - what, at a high level, are these components for?

Reply:
TBH we probably can get rid of the `Resolver` here, since we don't use it outside of the Ctor.

That said, the Formatter/Resolver system is fairly magical.

In General, a FormatterResolver is called via `.GetFormatter<T>()`. The contract is, either a proper `IMessageFormatter<T>` is returned, or `null` is returned. If `null` is returned at the end, well you can't serialize it. :)

`CompositeResolver`, will call each `IFormatterResolver` it is given from first to last, Until a value is returned. (It does cache internally, however there are perf hits due to a concurrentDictionary lookup.)